core/templates/services/search.service.ts

```diff
@@ -229,12 +230,16 @@ export class SearchService {
   }
 
   loadMoreData(
-      successCallback: (SearchResponseData, boolean) => void,
-      failureCallback?: (boolean) => void): void {
+      successCallback: (
+        SearchResponseData: SearchResponseBackendDict,
+        boolean: boolean) => void,
+      failureCallback?: (arg0: boolean) => void): void {
     // If a new query is still being sent, or the end of the page has been
     // reached, do not fetch more results.
     if (this._isCurrentlyFetchingResults || this.hasReachedEndOfPage()) {
-      failureCallback(this.hasReachedEndOfPage());
+      if (failureCallback) {
```

Comment:
should we throw error if the callback is not defined?

Reply:
I don't think we should. That is already an optional property. So we're just checking if the function is present then execute it, else return void.
Only use case of the function is 
![image](https://user-images.githubusercontent.com/60519303/128558897-a7d0a493-a4a2-47c9-bec0-67d4523446e4.png)